src/core/core.scale.js

```diff
@@ -1206,9 +1206,19 @@ export default class Scale extends Element {
       const color = optsAtIndex.color;
       const strokeColor = optsAtIndex.textStrokeColor;
       const strokeWidth = optsAtIndex.textStrokeWidth;
+      let tickTextAlign = textAlign;
 
       if (isHorizontal) {
         x = pixel;
+
+        if (textAlign === 'inner') {
+          if (i === ilen - 1) {
+            tickTextAlign = 'right';
+          } else if (i === 0) {
+            tickTextAlign = 'left';
+          }
+        }
```

Comment:
```suggestion
        if (textAlign === 'inner') {
          if (i === ilen - 1) {
            tickTextAlign = 'right';
          } else if (i === 0) {
            tickTextAlign = 'left';
          } else {
            tickTextAlign = 'center'
          }
        }
```

Reply:
This made me think what happens if the scale is reversed?

Reply:
Alignment also needs to be flipped in that case: 
![image](https://user-images.githubusercontent.com/39033624/152689397-613901d4-b60c-424b-a81e-15e4b1503c39.png)

Reply:
Thanks for this case investigation) Correct logic for reverse axis and add additional test